bpfd-operator/controllers/bpfd-agent/kprobe-program.go

```diff
@@ -208,7 +226,8 @@ func (r *KprobeProgramReconciler) reconcileBpfdProgram(ctx context.Context,
 
 	// BpfProgram exists but either KprobeProgram is being deleted or node is no
 	// longer selected....unload program
-	if isBeingDeleted || !isNodeSelected {
+	if isBeingDeleted || !isNodeSelected ||
+		(mapOwnerStatus.isSet && (!mapOwnerStatus.isFound || !mapOwnerStatus.isLoaded)) {
 		r.Logger.V(1).Info("KprobeProgram exists on Node but is scheduled for deletion or node is no longer selected", "isDeleted", isBeingDeleted,
```

Comment:
nit this needs to be updated... We'll have to go back and clean this up at some point but it's fine for now

Reply:
Done